extensions/amp-a4a/0.1/test/test-real-time-config-manager.js

```diff
@@ -0,0 +1,426 @@
+      'width': '200',
+      'height': '50',
+      'type': 'doubleclick',
+      'layout': 'fixed',
+    });
+    doc.body.appendChild(element);
+    fetchJsonStub = sandbox.stub(Xhr.prototype, 'fetchJson');
+    a4aElement = new AmpA4A(element);
+
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function setFetchJsonStubBehavior(params, response, isString, shouldFail) {
+    const textFunction = () => {
+      return !isString ? Promise.resolve(JSON.stringify(response)) :
+          Promise.resolve(response);
+    };
+    if (shouldFail) {
+      fetchJsonStub.withArgs(params).returns(Promise.reject());
+    } else {
+      fetchJsonStub.withArgs(params).returns(Promise.resolve({
+        status: 200,
+        text: textFunction,
+      }));
+    }
+  }
+
+  function setRtcConfig(rtcConfig) {
+    element.setAttribute('rtc-config', JSON.stringify(rtcConfig));
+  }
+
+  describe('#maybeExecuteRealTimeConfig_', () => {
+    function executeTest(args) {
+      const rtcConfig = {
+        urls: args.urls,
+        vendors: args.vendors,
+        timeoutMillis: args.timeoutMillis};
+      setRtcConfig(rtcConfig);
+      (args.expectedCalloutUrls || []).forEach((expectedUrl, i) => {
+        setFetchJsonStubBehavior(expectedUrl, args.rtcCalloutResponses[i],
+            args.responseIsString, args.failXhr);
+      });
+      const rtcResponsePromiseArray = maybeExecuteRealTimeConfig_(
+          a4aElement, args.customMacros);
+      return rtcResponsePromiseArray.then(rtcResponseArray => {
+        expect(fetchJsonStub.callCount).to.equal(args.calloutCount);
+        (args.expectedCalloutUrls || []).forEach(url => {
+          expect(fetchJsonStub.calledWith(url));
+        });
+        rtcResponseArray.forEach((rtcResponse, i) => {
+          expect(rtcResponse.rtcResponse).to.deep.equal(
+              args.expectedRtcArray[i].rtcResponse);
+          expect(rtcResponse.callout).to.equal(
+              args.expectedRtcArray[i].callout);
+          expect(rtcResponse.error).to.equal(args.expectedRtcArray[i].error);
+          expect(Object.keys(rtcResponse).sort()).to.deep.equal(
+              Object.keys(args.expectedRtcArray[i]).sort());
+          expect(isFiniteNumber(rtcResponse.rtcTime)).to.be.true;
+        });
+      });
+    }
+
+    const urlMacros = [
+      'SLOT_ID', 'PAGE_ID', 'ADX', 'ADY', 'WIDTH', 'HEIGHT',
+    ];
+
+    function generateUrls(numUrls, numMacroUrls) {
+      const urls = [];
+      for (let i = 0; i < numUrls; i++) {
+        urls.push(`https://www.${i}.com/`);
+      }
+      let url;
+      let macros;
+      for (let i = 0; i < numMacroUrls; i++) {
+        url = `https://www.${i + numUrls}.com/`;
+        macros = [];
+        for (let macroIndex = 0; macroIndex < i + 1; macroIndex++) {
+          macros.push(
+              `${urlMacros[macroIndex].toLowerCase()}=${urlMacros[macroIndex]}`
+          );
+        }
+        url += macros.join('&');
+        urls.push(url);
```

Comment:
No need to declare url variable at all.  Can just do:  urls.push(`https://www.${1 + numUrls}.com/${macros.join('&')}`);